pkg/datasources/account_roles.go

```diff
@@ -0,0 +1,94 @@
+package datasources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var accountRolesSchema = map[string]*schema.Schema{
+	"pattern": {
```

Comment:
so what i have done in some places is use 
```
like {
 pattern = "<pattern>"
}
```
The like doesn't really serve any purpose, but it does make it look more like the SQL. I don't have a problem either way, but we need to decide so we can be consistent in other data sources

Reply:
I think this PR was not meant to improve the resource/data source. It was to use the SDK from within the existing resource. One additional thing that @sfc-gh-jcieslak did was rename the resource to be consistent (but left the improvements for the upcoming resources rework).

Reply:
👍 Removed renamed version, but left new CRUD functions